<launch>
    <node pkg="suricate_robot" type="robot_twist_subscriber" name="robot_twist_subscriber" required="true" >
        <param name="max_pwm" type="int" value="255" />
        <param name="scaling_pwm" type="double" value="1" />
        <param name="topic_name" type="string" value="/robot/cmd_vel_real" />
        <param name="rate" type="int" value="100" />
    </node>

    <node pkg="suricate_robot" type="robot_imu_publisher.py" name="robot_imu_publisher" required="true" >
        <param name="offset_pitch" type="double" value="-8.01" />
        <param name="offset_roll" type="double" value="0" />
        <param name="offset_yaw" type="double" value="0" />
        <param name="rate" type="int" value="130" />
    </node>

    <node pkg="suricate_robot" type="robot_odometry.py" name="robot_odometry" required="true" >
        <param name="rate" type="int" value="150" />
        <param name="ticks_meter" type="double" value="15000" />
        <param name="base_width" type="double" value="0.6655" />
        <param name="encoder_min" type="double" value="-4294964113" />
        <param name="encoder_max" type="double" value="4294964113" />
    </node>

    <node pkg="robot_micro" type="robot_safety_checker.py" name="robot_safety_checker" required="true" >
        <param name="rate" type="int" value="50" />
    </node>

    <!--
    <node pkg="suricate_robot" type="robot_visualization.py" name="robot_visualization" required="true" />
     -->

    <!--
    <node pkg="suricate_robot" type="robot_controller.py" name="robot_controller" required="true" >
        <param name="S_kp" type="double" value="0.25" />
        <param name="S_kd" type="double" value="0.03" />
        <param name="S_ki" type="double" value="0.3" />
        <param name="S_windup" type="double" value="1" />

        <param name="P_kp" type="double" value="0.3" />
        <param name="P_kd" type="double" value="0.0" />
        <param name="P_ki" type="double" value="0.1" />
        <param name="P_windup" type="double" value="1" />

        <param name="scaling_joy" type="double" value="1" />
    </node>
    -->

    <!--
    <node pkg="robot_micro" type="robot_controller_LQR.py" name="robot_controller" required="true" ></node>
    -->
</launch>